src/platform/Linux/bluez/Helper.cpp

```diff
@@ -448,15 +445,14 @@ static gboolean BluezCharacteristicAcquireWrite(BluezGattCharacteristic1 * aChar
 #endif // CHIP_ERROR_LOGGING
         ChipLogError(DeviceLayer, "FAIL: socketpair: %s in %s", StringOrNullMarker(errStr), __func__);
         g_dbus_method_invocation_return_dbus_error(aInvocation, "org.bluez.Error.Failed", "FD creation failed");
-        goto exit;
+        return FALSE;
     }
 
-    options    = g_variant_dict_new(aOptions);
-    option_mtu = g_variant_dict_lookup_value(options, "mtu", G_VARIANT_TYPE_UINT16);
-    VerifyOrExit(
-        option_mtu != nullptr, ChipLogError(DeviceLayer, "FAIL: No MTU in options in %s", __func__);
+    option_mtu = GAutoPtr<GVariant>(g_variant_lookup_value(aOptions, "mtu", G_VARIANT_TYPE_UINT16));
```

Comment:
Use `g_variant_lookup()`, so there will be no `GVariant` to release. Also, check other places where `g_variant_dict_new()` is still used.